### PR TITLE
Fixed monkey cubes producing two monkeys when used on a sink.

### LINKED ISSUE #14667: Monkey cubes produce two monkeys
See title for bug description.

### PR DESCRIPTION
**Pull request by Buggy123**
Fixes #14667

Hopefully I didn't fuck it up.

### CODE REVIEW

**Comment by Menshin** [comment]
Well, independently of the bug this PR is supposed to fix, the code changed is obviously a simple copy-paste and some changes would be needed for this to even work (for example ```src``` is not the same thing when it's called by the sink object or by the cube one).

While maintainers (and other contributors) could help you with this on this page, it would be best (and simplier) to hop on the #coderbus IRC channel to discuss the matter.

File: code/game/objects/structures/watercloset.dm [hunk 1]
```diff
@@ -414,6 +414,16 @@
 		user << "<span class='warning'>Someone's already washing here!</span>"
 		return
 
+	if(istype(O, /obj/item/weapon/reagent_containers/food/snacks/monkeycube))
+		if (/obj/item/weapon/reagent_containers/food/snacks/monkeycube/wrapped)
```
Comment (Fox-McCloud) [inline code/game/objects/structures/watercloset.dm]: this seems a bit non-standard, especially in combination with the return /obj/item/weapon/reagent_containers/food/snacks/monkeycube/proc/Expand() below.

Usuallllyyy for stuff like this, you'd typecast to monkeycubes, then wreck for if they're wrapped or not (after all, maybe other monkey cubes, in the future, may be wrapped but not have this path), then you could also do x.expact() later on down.

File: code/game/objects/structures/watercloset.dm [hunk 2]
```diff
@@ -414,6 +414,16 @@
...
 		return
 
+	if(istype(O, /obj/item/weapon/reagent_containers/food/snacks/monkeycube))
+		if (/obj/item/weapon/reagent_containers/food/snacks/monkeycube/wrapped)
+			user << "<span class='notice'>You need to unwrap [src] first!</span>"
+			return
+		else
+			user << "<span class='notice'>You place [src] under a stream of water...</span>"
+			user.drop_item()
+			loc = get_turf(O)
```
Comment (Fox-McCloud) [inline code/game/objects/structures/watercloset.dm]: better to forcemove than setting loc directly.

File: code/game/objects/structures/watercloset.dm (hunk 1 above)
Comment (Menshin) [inline code/game/objects/structures/watercloset.dm]: Moreso when we want to check if ```O``` is wrapped (and not  checking if wrapped is a variable of monkeycude or whatever checking a typepath do).

File: code/game/objects/structures/watercloset.dm (hunk 2 above)
Comment (Menshin) [inline code/game/objects/structures/watercloset.dm]: Same for ```src``` used in the output stream.
Generally, this PR is simply code moving for now and needs to be cleaned up.

File: code/game/objects/structures/watercloset.dm (hunk 1 above)
Comment (tkdrg) [inline code/game/objects/structures/watercloset.dm]: I didn't even know checking a typepath compiled (it really shouldn't). And yeah, handling this on an istype() in sinks is bad, afterattack is better encapsulation.

File: code/game/objects/structures/watercloset.dm (hunk 2 above)
Comment (tkdrg) [inline code/game/objects/structures/watercloset.dm]: Yeah, what menshin said. Just copy-pasting this from the right place to this wrong one won't fix anything, won't even work, and will introduce even more issues.

File: code/game/objects/structures/watercloset.dm [hunk 3]
```diff
@@ -414,6 +414,16 @@
...
+			return /obj/item/weapon/reagent_containers/food/snacks/monkeycube/proc/Expand()
```
Comment (tkdrg) [inline code/game/objects/structures/watercloset.dm]: This won't work the way you seem to expect.

**Comment by Buggy123** [comment]
Wow i should've checked this sooner too. Welp, my code is shit. I have no idea how to fix it, the best I can do is copypasta and edit it slightly.

**Comment by tkdrg** [comment]
It's not even creating two monkeys for me as I test this. @ChangelingRain can you check if this is still happening?

**Comment by duncathan** [comment]
I think this one might need another attempt

**Pull request closed without merging**